pkg/usersettings/helpers.go

```diff
@@ -0,0 +1,95 @@
+package usersettings
+
+import (
+	"errors"
+
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func getUserSettingMeta(name string, uid string) (*UserSettingMeta, error) {
+	resourceIdentifier := ""
+
+	if name == "kube:admin" {
+		resourceIdentifier = "kubeadmin"
+	} else if uid != "" {
+		resourceIdentifier = uid
```

Comment:
Make sure there is no UID for the kubeadmin case:
```suggestion
	if uid != "" {
		resourceIdentifier = uid
	} else if name == "kube:admin" {
		resourceIdentifier = "kubeadmin"
```

Reply:
Changed.